codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/GenerationContext.java

```diff
@@ -161,4 +164,25 @@ public List<RubyRuntimePlugin> getRuntimePlugins() {
                 .flatMap(List::stream)
                 .collect(Collectors.toList());
     }
+
+    /**
+     * @return list of all AuthSchemes from all integrations and the application transport.
+     */
+    public List<AuthScheme> getAuthSchemes() {
+        List<AuthScheme> authSchemes = applicationTransport().defaultAuthSchemes();
+        authSchemes.add(AnonymousAuthSchemeFactory.build());
+        integrations().forEach((i) -> {
+            i.getAdditionalAuthSchemes(this).forEach((s) -> authSchemes.add(s));
+        });
+
+        ServiceIndex.of(model()).getAuthSchemes(service()).forEach((shapeId, trait) -> {
+            AuthScheme authScheme = authSchemes.stream()
+                    .filter(s -> s.getShapeId().equals(shapeId))
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalStateException("No auth scheme found for " + shapeId));
+            authScheme.extractSignerProperties(trait);
```

Comment:
Why is this done here?  Is this modifying the auth scheme?  It seems like the authScheme itself shouldn't be modifiable once returned.

Reply:
Yeah, it was modifying auth scheme. I added trait to the getter as you suggested, and I need to merge the static properties from the builder each time the getter is called.